Lib/sqlite3/test/hooks.py

```diff
@@ -259,6 +259,8 @@ def trace(statement):
         cur.execute(queries[0])
         con2.execute("create table bar(x)")
         cur.execute(queries[1])
+        con1.close()
+        con2.close()
```

Comment:
May use try/finally to ensure that the connection is closed if an assertion fails. Or use an helper function.

Reply:
Good point

Reply:
I added d31403eb9885eb4654741a84b28f5dcd7179acad.